RadixWallet/Clients/CloudBackupClient/CloudBackupClient+Live.swift

```diff
@@ -123,59 +120,99 @@ extension CloudBackupClient {
 		}
 
 		@Sendable
-		func backupProfileAndSaveResult(_ profile: Profile) async {
-			let existingRecord = try? await fetchProfileRecord(.init(recordName: profile.id.uuidString))
-			let result: BackupResult.Result
+		func backupProfileAndSaveResult(_ profile: Profile, existingRecord: CKRecord?) async throws {
+			try? userDefaults.setLastCloudBackup(.started(.now), of: profile)
+
 			do {
 				let json = profile.toJSONString()
-				try await uploadProfileToICloud(.right(json), header: profile.header, existingRecord: existingRecord)
-				result = .success
-			} catch CKError.accountTemporarilyUnavailable {
-				result = .temporarilyUnavailable
-			} catch CKError.notAuthenticated {
-				result = .notAuthenticated
+				try await backupProfile(.right(json), header: profile.header, existingRecord: existingRecord)
 			} catch {
-				loggerGlobal.error("Automatic cloud backup failed with error \(error)")
-				result = .failure
+				let failure: BackupResult.Result.Failure
+				switch error {
+				case CKError.accountTemporarilyUnavailable:
+					failure = .temporarilyUnavailable
+				case CKError.notAuthenticated:
+					failure = .notAuthenticated
+				default:
+					loggerGlobal.error("Automatic cloud backup failed with error \(error)")
+					failure = .other
+				}
+
+				try? userDefaults.setLastCloudBackup(.failure(failure), of: profile)
+				throw error
 			}
 
-			try? userDefaults.setLastCloudBackup(result, of: profile)
+			try? userDefaults.setLastCloudBackup(.success, of: profile)
 		}
 
+		@Sendable
+		func performAutomaticBackup(_ profile: Profile, timeToCheckIfClaimed: Bool) async {
+			let needsBackUp = profile.appPreferences.security.isCloudProfileSyncEnabled && profile.header.isNonEmpty
+			let existingRecord = try? await fetchProfileRecord(profile.id)
+			let backedUpHeader = try? existingRecord.map(getProfileHeader)
+			let isBackedUp = backedUpHeader?.saveIdentifier == profile.header.saveIdentifier
+			let shouldBackUp = needsBackUp && !isBackedUp
+
+			guard shouldBackUp || timeToCheckIfClaimed else { return }
+
+			let shouldReclaim: Bool
+			if let backedUpID = backedUpHeader?.lastUsedOnDevice.id, await !profileStore.isThisDevice(deviceID: backedUpID) {
+				let action = await overlayWindowClient.scheduleFullScreen(.init(root: .claimWallet(.init())))
+				switch action {
+				case .claimWallet(.transferBack):
+					shouldReclaim = true
+				case .claimWallet(.didClearWallet), .dismiss:
```

Comment:
I would leave the `assertionFailure` for the `.dismiss` case. It isn't possible right now, but if at some point we add a `X` to every full screen or something, we better fail rather than leave the wallet on an unwanted case (wallet not claimed nor deleted).

Reply:
It does actually sometimes get sent, about once in 25 presentations, in addition to the actual action. This is because it's the fallback action if the actions sequence completes prematurely, which seems to happen on rare occasions. It's not visible to the user so we better postpone that investigation.

Reply:
but is the view actually dismissed?

Reply:
Yes, there is no user-visible glitch